E2guardian first package files

marcelloc opened the pull request:
e2guardian - first files

marcelloc committed:
e2guardian - first files

marcelloc committed:
e2guardian - package info

marcelloc committed:
e2guardian - fix copyright info

phil-davis commented:
I guess a lot of these files came from DansGuardian package and have been modified for E2guardian.
I notice they have lots of "compressed" syntax that does not match the normal code style for pfSense, like:
$xyz=$abc;
if($condition){}
"abc"=>"xyz"
I could do a bunch of find-replace on this stuff to make it match current code style - it does not take that long with some decent regex. Does anyone care enough?

phil-davis commented:
And did you want to remove all traces of DansGuardian in the code? e.g.
function dg_text_area_decode() ...
and var names like $dans_group_source
I don't mind massaging stuff like this while in evening zombie mode with the TV.

phil-davis commented on config/e2guardian/e2guardian.inc:
```diff
@@ -0,0 +1,1458 @@
+		$e2guardian_blacklist=$config['installedpackages']['e2guardianblacklist']['config'][0];
+
+	#daemon options
+	$e2guardian_enabled=$e2guardian['enable_dg'];
```
This seems odd - and I don't see enable_dg or $e2guardian_enabled anywhere else here.

phil-davis commented on config/e2guardian/e2guardian.inc:
```diff
@@ -0,0 +1,1458 @@
+	$contentscanexceptions=($e2guardian_config['contentscanexceptions']?"on":"off");
+	$recheckreplacedurls=(preg_match('/recheckreplacedurls/',$e2guardian_config['misc_options'])?"on":"off");
+	$forwardedfor=(preg_match('/forwardedfor/',$e2guardian_config['misc_options'])?"on":"off");
+	$recheckreplacedurls=(preg_match('/icapscan/',$e2guardian_config['misc_options'])?"on":"off");
```
This looks odd - "icapscan" result goes into $recheckreplacedurls - both of these are already on lines above.

phil-davis commented on config/e2guardian/e2guardian.inc:
```diff
@@ -0,0 +1,1458 @@
+			file_notice("e2guardian - " . $e2guardian_log['reportingcgi'] . " is not a valid access denied cgi url","");
+			}
+	}
+	$accessdenied=($e2guardian_log['reportingcgi']?$e2guardian_log['report_level']:"3");
```
This looks odd, and I don't see any other use of $accessdenied

phil-davis commented on config/e2guardian/e2guardian.inc:
```diff
@@ -0,0 +1,1458 @@
+					$groupaccessdeniedaddress="accessdeniedaddress = '".$cgimatches[1]."'";
+					}
+				else{
+					log_error('E2guardian - Group '.$e2guardian_groups['name']. ' does not has a valid access denied cgi url.');
```
s/not has/not have/

phil-davis commented on config/e2guardian/e2guardian.inc:
```diff
@@ -0,0 +1,1458 @@
+					}
+				else{
+					log_error('E2guardian - Group '.$e2guardian_groups['name']. ' does not has a valid access denied cgi url.');
+					file_notice('E2guardian - Group '.$e2guardian_groups['name']. ' does not has a valid access denied cgi url.',"");
```
s/not has/not have/

phil-davis commented on config/e2guardian/e2guardian.inc:
```diff
@@ -0,0 +1,1458 @@
+					${$list_value}=array_merge(explode("\n",$file_temp),${$list_value});
+				}
+				#add a package warning
+				array_unshift(${$list_value},"#Do not edit this file.","#It's created by e2guardian package and overwrited every config save.");
```
s/overwrited/overwritten/

phil-davis commented on config/e2guardian/e2guardian.php:
```diff
@@ -0,0 +1,180 @@
+	global $config,$g;
+	if (is_array($config['installedpackages']['e2guardianblacklist']) && is_array($config['installedpackages']['e2guardianblacklist']['config'])){
+	   	$url=$config['installedpackages']['e2guardianblacklist']['config'][0]['url'];
+	   	$uw="Found a previouns install, checking Blacklist config...";
```
s/previouns/previous/

phil-davis commented on config/e2guardian/e2guardian.xml:
```diff
@@ -0,0 +1,387 @@
+		<field>
+			<fielddescr>Listen Interface(s)</fielddescr>
+			<fieldname>interface</fieldname>
+			<description><![CDATA[Default: <strong>LAN/loopback</strong><br>Select interface(s) that you want to e2guardian listen on.<br>
```
e2guardian to listen on

phil-davis commented on config/e2guardian/e2guardian.xml:
```diff
@@ -0,0 +1,387 @@
+			<type>input</type>
+			<size>10</size>
+			<description><![CDATA[Default: <strong>4/32</strong><br>
+			 				 Sets the minimum and maximun number of processes to be kept ready to handle connections.<br>
```
s/maximun/maximum/

phil-davis commented on config/e2guardian/e2guardian.xml:
```diff
@@ -0,0 +1,387 @@
+			<type>input</type>
+			<size>10</size>
+			<description><![CDATA[Default: <strong>0</strong><br>
+			 				Sets the maximum number client IP addresses allowed to connect at once.<br>
```
number of client IP...

phil-davis commented on config/e2guardian/e2guardian.xml:
```diff
@@ -0,0 +1,387 @@
+		<field>
+			<fielddescr>Proxy Time out</fielddescr>
+			<fieldname>proxytimeout</fieldname>
+			<description><![CDATA[This option handle max time to wait for proxy server.]]></description>
```
s/handle/handles/

phil-davis commented on config/e2guardian/e2guardian_about.php:
```diff
@@ -0,0 +1,110 @@
+require_once("guiconfig.inc");
+
+$pfSversion = str_replace("\n", "", file_get_contents("/etc/version"));
+if(strstr($pfSversion, "1.2"))
```
I guess this new package is only going to be supported from 2.2.* onwards. So this 1.2 stuff could be removed. This test also looks like it would fail to distinguish between pfSense 1.2.* and 2.1.2

phil-davis commented on config/e2guardian/e2guardian_about.php:
```diff
@@ -0,0 +1,110 @@
+                        </tr>
+						<tr>
+                        <td width="22%" valign="top" class="vncell"><?=gettext("Whatis");?></td>
+						<td width="78%" class="vtable"><?=gettext("<a target=_new href='http://e2guardian.org/?page=whatisdg'>What is E2guardian</a><br><br>");?>
```
page=whatisdg does happen to redirect to a general page about E2guardian - but I am not sure that leaving a "dg" reference here is intended?

phil-davis commented on config/e2guardian/e2guardian_antivirus_acl.xml:
```diff
@@ -0,0 +1,241 @@
+			<tab_level>2</tab_level>
+		</tab>
+		<tab>
+			<text>Searche Engine</text>
```
s/Searche/Search/
might be in other places also

phil-davis commented on config/e2guardian/e2guardian_blacklist.xml:
```diff
@@ -0,0 +1,173 @@
+		<field>
+			<fielddescr>List on</fielddescr>
+			<fieldname>liston</fieldname>
+			<description><![CDATA[Select where to list categories dowloaded form file<br>
```
s/dowloaded/downloaded/
should "form" be "from".
and then I am still not sure what this means.

phil-davis commented on config/e2guardian/e2guardian_blacklist.xml:
```diff
@@ -0,0 +1,173 @@
+			<fielddescr>List on</fielddescr>
+			<fieldname>liston</fieldname>
+			<description><![CDATA[Select where to list categories dowloaded form file<br>
+							List on Banned and Exception is usefull if you want to whitelist any category downloaded from url.]]></description>
```
s/usefull/useful/

phil-davis commented on config/e2guardian/e2guardian_config.xml:
```diff
@@ -0,0 +1,316 @@
+		<field>
+			<fielddescr>Auth Plugins</fielddescr>
+			<fieldname>auth_plugin</fieldname>
+			<description><![CDATA[This option handle the extraction of client usernames from various sources, such as Proxy-Authorisation headers and ident servers, 
```
s/handle/handles/

phil-davis commented on config/e2guardian/e2guardian_config.xml:
```diff
@@ -0,0 +1,316 @@
+			<type>input</type>
+			<size>10</size>
+			<description><![CDATA[Positive (clean) result caching for URLs Caches good pages so they don't need to be scanned again.It also works with AV plugins.<br>
+			0 = off (recommended for ISPs with users with disimilar browsing)<br>
```
s/disimilar/dissimilar/

phil-davis commented on config/e2guardian/e2guardian_file_acl.xml:
```diff
@@ -0,0 +1,249 @@
+		<field>
+			<fielddescr>Extension</fielddescr>
+			<fieldname>exception_extensionlist</fieldname>
+			<description><![CDATA[Enter banned extension List for this group.<br>	
```
s/banned/exception/

phil-davis commented on config/e2guardian/e2guardian_file_acl.xml:
```diff
@@ -0,0 +1,249 @@
+		<field>
+			<fielddescr>Mime</fielddescr>
+			<fieldname>exception_mimetypelist</fieldname>
+			<description><![CDATA[Enter banned mimetype List for this group.<br>	
```
s/banned/exception/

phil-davis commented on config/e2guardian/e2guardian_groups.xml:
```diff
@@ -0,0 +1,460 @@
+			<size>60</size>
+		</field>
+		<field>
+			<name>Acess Lists</name>
```
s/Acess/Access/

phil-davis commented on config/e2guardian/e2guardian_groups.xml:
```diff
@@ -0,0 +1,460 @@
+		<field>
+			<fielddescr>Naughtiness limit</fielddescr>
+			<fieldname>naughtynesslimit</fieldname>
+			<description><![CDATA[This the limit over which the page will be blocked.  Each weighted phrase is given a value either positive or negative and the values added up.<br>
```
This is the limit...

phil-davis commented on config/e2guardian/e2guardian_groups.xml:
```diff
@@ -0,0 +1,460 @@
+			<fielddescr>LDAP</fielddescr>
+			<fieldname>ldap</fieldname>
+			<description><![CDATA[Select LDAP servers to extract users from<br>
+								 The group must has the same name( or description) in e2guardian and on active directory<br>
```
s/has/have/

phil-davis commented on config/e2guardian/e2guardian_groups.xml:
```diff
@@ -0,0 +1,460 @@
+ 				<option><name>Password CANNOT Change(code 64)</name><value>64</value></option>
+ 				<option><name>Password has Expired (code 8388608)</name><value>8388608</value></option>
+ 				<option><name>Account will Never Expire (code 65536)</name><value>65536</value></option>
+ 				<option><name>Enabled and Does NOT expire Paswword (code 66048)</name><value>66048</value></option>
```
s/Paswword/Password/

phil-davis commented on config/e2guardian/e2guardian_groups.xml:
```diff
@@ -0,0 +1,460 @@
+		<field>
+			<fielddescr>Update frequency</fielddescr>
+			<fieldname>freq</fieldname>
+			<description><![CDATA[How often extract users from active directory and verify changes<br>
```
How often to extract...

phil-davis commented on config/e2guardian/e2guardian_ips.xml:
```diff
@@ -0,0 +1,141 @@
+		<field>
+			<fieldname>exceptioniplist</fieldname>
+			<fielddescr>Exception Ip List</fielddescr>
+			<description><![CDATA[Include ip addresses and or ipadresses/netmask of computers from which web access should not be filtered.<br>
```
s/adresses/addresses/

phil-davis commented on config/e2guardian/e2guardian_ips.xml:
```diff
@@ -0,0 +1,141 @@
+		<field>
+			<fieldname>default</fieldname>
+			<fielddescr>Default</fielddescr>
+			<description><![CDATA[Include ip addresses and or ipadresses/netmask for this group one per line<br>Hint:192.168.1.0/255.255.255.0<br>192.168.1.5]]></description>
```
s/adresses/addresses/

phil-davis commented on config/e2guardian/e2guardian_ips_header.template:
```diff
@@ -0,0 +1,113 @@
+		<field>
+			<fieldname>exceptioniplist</fieldname>
+			<fielddescr>Exception Ip List</fielddescr>
+			<description><![CDATA[Include ip addresses and or ipadresses/netmask of computers from which web access should not be filtered.<br>
```
s/adresses/addresses/

phil-davis commented on config/e2guardian/e2guardian_ldap.xml:
```diff
@@ -0,0 +1,174 @@
+	</field>
+	<field>
+		<fielddescr>mask</fielddescr>
+		<description><![CDATA[Default: USER<br>Apply prefix or sufix any info to username extracted.<br>
```
s/sufix/suffix/

phil-davis commented on config/e2guardian/e2guardian_limits.xml:
```diff
@@ -0,0 +1,183 @@
+			<type>input</type>
+			<size>10</size>
+			<description><![CDATA[<strong>Default is 2000</strong><br>
+						This is only used if you use a content scanner plugin such as AV. This is the max size of file that DG will download and cache in RAM.<br>
```
s/DG/E2/

phil-davis commented on config/e2guardian/e2guardian_limits.xml:
```diff
@@ -0,0 +1,183 @@
+			<type>input</type>
+			<size>10</size>
+			<description><![CDATA[<strong>Default is 2000</strong><br>
+						 This is only used if you use a content scanner plugin such as AV. This is the max size file that DG will download so that it can be scanned or virus checked.<br>
```
s/DG/E2/

phil-davis commented on config/e2guardian/e2guardian_phrase_acl.xml:
```diff
@@ -0,0 +1,272 @@
+		<field>
+			<fielddescr>Config file</fielddescr>
+			<fieldname>banned_phraselist</fieldname>
+			<description><![CDATA[Enter Banned Pharase List for this group.<br>	
```
s/Pharase/Phrase/

phil-davis commented on config/e2guardian/e2guardian_url_acl.xml:
```diff
@@ -0,0 +1,353 @@
+		<field>
+			<fielddescr>config</fielddescr>
+			<fieldname>modify_urllist</fieldname>
+			<description><![CDATA[Enter Exception url List for this group.<br>	
```
This text seems wrong? Maybe "modify URL list", whatever that means?

phil-davis commented on config/e2guardian/e2guardianfx.conf.template:
```diff
@@ -0,0 +1,388 @@
+disablecontentscan = {$e2guardian_groups['disablecontentscan']}
+
+# Enable Deep URL Analysis
+# When enabled, DG looks for URLs within URLs, checking against the bannedsitelist and
```
s/DG/E2/

phil-davis commented on config/e2guardian/e2guardianfx.conf.template:
```diff
@@ -0,0 +1,388 @@
+# only used in reporting level 3.
+#
+# The default template file path is <languagedir>/<language>/template.html
+# e.g. {$dg_dir}/share/e2guardian/languages/ukenglish/template.html when using 'ukenglish'
```
what to do about old references to dg like this $dg_dir ?

phil-davis commented on config/e2guardian/e2guardianfx.conf.template:
```diff
@@ -0,0 +1,388 @@
+# this is in seconds. If 'violations' occur in 'threshold' seconds, then
+# a notification is made.
+# if this is set to 0, then whenever the set number of violations are made a 
+# notifaction will be sent.
```
s/notifaction/notification/

marcelloc committed:
e2guardian -fix a lot of typos and old references do dansguardian

marcelloc commented:
thanks phil-davis, I've fixed all typos and removed most references do dansguardian

phil-davis commented on config/e2guardian/e2guardian_phrase_acl.xml:
```diff
@@ -0,0 +1,272 @@
+		<field>
+			<fielddescr>Config file</fielddescr>
+			<fieldname>weighted_phraselist</fieldname>
+			<description><![CDATA[Enter Weighted Pharase List for this group.<br>	
```
s/Pharase/Phrase/

phil-davis commented on config/e2guardian/e2guardian_phrase_acl.xml:
```diff
@@ -0,0 +1,272 @@
+		<field>
+			<fielddescr>Config file</fielddescr>
+			<fieldname>exception_phraselist</fieldname>
+			<description><![CDATA[Enter exception Pharase List for this group.<br>	
```
s/Pharase/Phrase/

marcelloc committed:
e2guardian - fix identation

marcelloc committed:
e2guardian - fix identation

rbgarga commented on pkg_config.10.xml:
```diff
@@ -431,27 +431,26 @@
+			<port>www/e2guardian</port>
 		</build_pbi>
-		<build_options>dansguardian_UNSET_FORCE=APACHE;dansguardian_SET_FORCE=TRICKLE CLAMD ICAP NTLM SSL</build_options>
+		<build_options>dansguardian_UNSET_FORCE=APACHE;dansguardian_SET_FORCE=TRICKLE CLAMD ICAP NTLM</build_options>
```
options are still referencing dansguardian. BTW, IMO it would be better to keep dansguardian entry as it is and add a new entry for e2guardian

rbgarga commented on config/e2guardian/e2guardian.php:
```diff
@@ -0,0 +1,185 @@
+	$dir = E2GUARDIAN_DIR . "/etc/e2guardian/lists";
+	//read e2guardian lists dirs
+	$groups = array("phraselists", "blacklists", "whitelists");
+	/ssigns know list files
```
You probably forgot one more / on this comment, it breaks syntax

rbgarga commented:
The following files are present on this PR, but are not listed as <additional_files_needed>, then they won't be installed:

e2guardian_ips.xml
e2guardian_rules_version.php
e2guardian_users.xml

rbgarga commented on config/e2guardian/e2guardian.inc:
```diff
@@ -0,0 +1,1471 @@
+require_once("util.inc");
+require_once("globals.inc");
+#require("guiconfig.inc");
+//$shortcut_section = "squid";
```
Why keep these comments here? It's a new file, if it's not necessary, just drop the line

rbgarga commented on config/e2guardian/e2guardian.inc:
```diff
@@ -0,0 +1,1471 @@
+
+require_once("util.inc");
+require_once("globals.inc");
+#require("guiconfig.inc");
```
Please do not use # for comments, use // or /* */

rbgarga commented on config/e2guardian/e2guardian.inc:
```diff
@@ -0,0 +1,1471 @@
+
+	#check certificates
+	$cert_count = 0;
+	if (is_dir('/usr/local/share/certs')){
```
Please respect style guide, add a space before {

rbgarga commented on config/e2guardian/e2guardian.inc:
```diff
@@ -0,0 +1,1471 @@
+        		}
+    		}
+        }
+	}
```
Please fix indent

marcelloc committed:
e2guardian - more ident fixes and remove files from install

marcelloc committed:
e2guardian -remove dansguardian references form pkg_config.10.xml

marcelloc commented:
Can it me merged?

rbgarga commented:
The code needs to follow developer style guide, there are a lot of places it doesn't.

marcelloc commented:
I've revised it two times after all typo revisions. Do you really think it
still has cosmetic mistakes that prevents it to be merged?
Em 11/06/2015 14:59, "Renato Botelho" <notifications@github.com> escreveu:

> The code needs to follow developer style guide, there are a lot of places
> it doesn't.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pfsense/pfsense-packages/pull/866#issuecomment-111223957>
> .
>

rbgarga commented:
Yeah, it has. There are problems like:

* tabs and spaces mixed (e.g. line 372 of e2guardian.inc but there are more)
* Bad indent (e.g. line 383 of e2guardian.inc)
* Spaces or tabs at EOL
* Lines without space between ')' and '{' (over 200 references on a quick grep)

After the code is complaint with developer style guide I can start testing it, but we cannot add new code what is not complaint

phil-davis commented:
@marcelloc I am happy to "developer style guide" this, that will save you a heap of time. I have done e2guardian.inc just now (many 100s of search/replace!). That is the big one. I will send a pull request against your repo/branch maybe by end of today.
Then you might like to make a fresh copy of everything, put it in a new branch, and submit a new clean pull request with the "final" files.

phil-davis commented on config/e2guardian/e2guardian.php:
```diff
@@ -0,0 +1,185 @@
+									if (!preg_match ("/^\./",$file)){
+										//assign list to array
+										$type = split("_",$file);
+										if (preg_match("/(\w+)/",$type[0],$matches));
```
The semi-colon on the end of this line makes a bug. The "if" will terminate there, and the next line will be executed regardless of the "if".
I have fixed in my code style review that I am doing now.

Commit:
E2Guardian Code Style

phil-davis commented:
@marcelloc I have sent you a pull request with code style changes.

marcelloc committed:
Merge pull request #1 from phil-davis/e2g-review

E2Guardian Code Style

marcelloc commented:
Hi Garga, with phill's help, can this code be revised?

cbuechler commented:
Renato's out on vacation at the moment, he'll get back next week and review.

Commit:
e2guardian.xml - remove useless tags, fix permissions, cosmetics

Commit:
e2guardian.conf.template - fix typos, improve English, make more readable

Attempt to improve readability and use consistent style in the configuration file. Also, many English improvements.
You might want to submit this upstream (https://github.com/e2guardian/e2guardian/blob/master/configs/e2guardian.conf.in) to make future maintenance easier. 

Basically, trying to use the following structure consistently:

# Short Setting Variable Description Capitalized
#
# Detailed descriptions - first paragraph...
#
# Details description - second paragraph (separated by a single line) Etc..
#disabledetting = 'value'
enabledsetting = 'value'

NOTE: There still are some traces of DG in variables used, I did not touch those to not break things - $dg, ipcfilename, urlipcfilenamem ipipcfilename. Please, review.

Commit:
e2guardian.conf.template - fix spell-checker induced typos.

E2guarding -> E2guardian.

Commit:
e2guardian.inc - code fixes

- Remove unused $pf_version
- Fix comments style (# -> //)
- dg_get_real_interface_address() -> e2_get_real_interface_address()
- use (platform_booting() for boot process check

Commit:
e2guardian.php - cosmetic fixes

- Fix pfSense website link
- Consistent comments style
- Full path to /bin/rm

Commit:
e2guardian_about.php - code style fixes

- Fix pfSense website link
- Fix broken indentation in copyright header
- XHTML compliance fixes

Commit:
e2guardian_antivirus_acl.xml - code style fixes

- Fix pfSense website in header
- Remove useless unused tags
- XHTML

Commit:
e2guardian_blacklist.xml - code style fixes

- Fix pfSense website in header
- Remove useless unused tags
- XHTML

Commit:
e2guardian_config.xml - code style and other fixes

- Fix pfSense website link.
- Remove useless unused tags
- Make the descriptions wrapped in CDATA actually easily readable and editable.
- XHTML
- Use consistent casing in options names
- Use descriptions consistent with and taken from e2guardian.conf.template where possible.

Commit:
e2guardian_blacklist.xml - revisit CDATA descriptions

- Make the descriptions wrapped in CDATA actually easily readable and editable.
- Use consistent casing in options names.

Commit:
e2guardian_antivirus_acl.xml -revisit CDATA descriptions

Make the descriptions wrapped in CDATA actually easily readable and editable.

Commit:
e2guardian_content_acl.xml - code style fixes

- Fix pfSense website link.
- Remove useless unused tags
- Make the descriptions wrapped in CDATA actually easily readable and editable.
- XHTML

Commit:
e2guardian_file_acl.xml - code style fixes

- Fix pfSense website link.
- Remove useless unused tags
- Make the descriptions wrapped in CDATA actually easily readable and editable.
- XHTML
- Consistent casing

Commit:
e2guardian_groups.xml - code style fixes

- Fix pfSense website link.
- Remove useless unused tags
- Nuke bunch of useless CDATA wraps.
- Make the remaining descriptions wrapped in CDATA actually easily readable and editable.
- XHTML
- Use consistent casing in options names
- Use descriptions consistent with and taken from e2guardian.conf.template where possible.

Commit:
e2guardian_header_acl.xml - code style fixes

- Fix pfSense website link.
- Remove useless unused tags
- Make the descriptions wrapped in CDATA actually easily readable and editable.
- XHTML
- Consistent casing

Commit:
e2guardian_ips_header.template - code style fixes

- Fix filename and pfSense website link in header
- Remove pointless unused tags
- Make the descriptions wrapped in CDATA actually easily readable and editable
- XHTML

Commit:
e2guardian_ldap.php - code style fixes

- Fix pfSense website link in header
- Fix indentation
- Consistent comments style

Commit:
e2guardian_ldap.xml - code style fixes

- Fix pfSense website link.
- Remove useless unused tags
- Make the descriptions wrapped in CDATA actually easily readable and editable.
- XHTML
- Consistent casing

Commit:
e2guardian_limits.xml - code style fixes

- Fix pfSense website link.
- Remove useless unused tags
- Make the descriptions wrapped in CDATA actually easily readable and editable.
-  Use descriptions consistent with and taken from e2guardian.conf.template where possible.
- XHTML
- Consistent casing

Commit:
e2guardian_log.xml - code style and other fixes

- Fix pfSense website link.
- Remove useless unused tags
- Use descriptive language names and reorder the languages list
- Nuke bunch of useless CDATA wraps.
- Make the descriptions wrapped in CDATA actually easily readable and editable.
- Use descriptions consistent with and taken from e2guardian.conf.template where possible.
- XHTML
- Consistent casing

Commit:
e2guardian_phrase_acl.xml - code style fixes

- Fix pfSense website link.
- Remove useless unused tags
- Nuke bunch of useless CDATA wraps.
- Make the descriptions wrapped in CDATA actually easily readable and editable.
- XHTML
- Consistent casing

Commit:
e2guardian_pics_acl.xml - code style fixes

- Fix pfSense website link.
- Remove useless unused tags
- Make the descriptions wrapped in CDATA actually easily readable and editable.
- XHTML
- Consistent casing

Commit:
e2guardian_search_acl.xml - codes style fixes

- Fix pfSense website link.
- Remove useless unused tags
- Nuke bunch of useless CDATA wraps
- XHTML
- Consistent casing

Commit:
e2guardian_site_acl.xml - code style fixes

- Fix pfSense website link.
- Remove useless unused tags
- Nuke bunch of useless CDATA wraps
- Make the descriptions wrapped in CDATA actually easily readable and editable
- XHTML
- Consistent casing
- Consistent casing

Commit:
e2guardian_sync.xml - code style fixes

- Fix pfSense website link.
- Remove useless unused tags
- Consistent casing

marcelloc committed:
Merge pull request #2 from doktornotor/patch-3

e2guardian - code style and improvements
Thanks doktornotor

Commit:
e2guardian - code style and improvements

- Fix pfSense website link.
- Remove useless unused tags
- Nuke bunch of useless CDATA wraps.
- Make the descriptions wrapped in CDATA actually easily readable and editable.
- XHTML
- Consistent casing

Commit:
e2guardian_users_header.template - code style fixes

- Fix pfSense website link
- Remove useless unused tags

Commit:
e2guardianfx.conf.template - fix typos, improve English, make more readable

Attempt to improve readability and use consistent style in the configuration file. Also, many English improvements.
You might want to submit this upstream (https://github.com/e2guardian/e2guardian/blob/master/configs/e2guardianf1.conf.in) to make future maintenance easier. 

Basically, trying to use the following structure consistently:

# Short Setting Variable Description Capitalized
#
# Detailed descriptions - first paragraph...
#
# Details description - second paragraph (separated by a single line) Etc..
#disabledetting = 'value'
enabledsetting = 'value'

Commit:
pkg_e2guardian.inc - fix pfSense website link

marcelloc committed:
Merge pull request #4 from doktornotor/patch-5

e2guardian - code style and improvements (the rest)

rbgarga commented:
building PBIs...

rbgarga commented:
@marcelloc PBIs are built, can you try them to check if everything is OK?

http://files.pfsense.org/packages/10/All/e2guardian-3.0.4_1-amd64.pbi
http://files.pfsense.org/packages/10/All/e2guardian-3.0.4_1-i386.pbi

If everything is OK, I'll proceed with the merge, changing required_version to 3.0 to not show it yet for users, so you can make one more test of install/deinstall, and all other checks before it goes public

aaronouthier commented:
Marcello,
Did you have a chance to test and review these?

If not, I would be happy to try them out. I have a system running VirtualBox, as well as a production router running x64 pfsense.

I have already tried adding the x64 package via command line, but I am not getting the GUI components. I looked into why, and it appears that this is normal when installing from a command line. I am happy to help, but I'll need the secret to installing a non-standard package from the WebUI.

deajan commented:
+1, I'm playing with some virtual pfsense instances right now and would not mind testing these before going into prod.

rbgarga commented:
@marcelloc did you get a chance to check the PBIs?

Commit:
XMLRPC sync improvements

Commit:
XMLRPC sync fixes and improvements

doktornotor commented:
@marcelloc 
I did a pull request regarding XMLRPC sync fixes. Please, check.

Another thing: 

Can we please get rid of ALL antivirus stuff for now (I got this working in Squid 3.4, this was an immense amount of work. Installing E2G with Squid as it is will
- install duplicate clamav service
- damage the clamav directory permissions (/var/run/clamav, /var/db/clamav, /var/log/clamav)
- there are numerous other problems identical to the old Squid3 package with this stuff

marcelloc committed:
Merge pull request #6 from doktornotor/patch-3

XMLRPC sync fixes and improvements

sunweaver commented:
Hi all, thanks to the work on e2guardian.

Now that pfSense 2.3.x is out and (as it seems) DansGuardian got removed from hit (yeah, it indeed is highly unmaintained by upstream...), I wonder if anything else is needed to merge this PR into pfsense/pfsense-package's master branch?

Thanks for feedback,
Mike

rbgarga commented:
This repository is obsolete, it was used only for pfSense 2.2 and earlier versions.

For current pfSense versions, packages are hosted at https://github.com/pfsense/freebsd-ports
